PWGCF/Femto3D/TableProducer/singleTrackSelector.cxx

```diff
@@ -0,0 +1,180 @@
+    } else {
+      grpmag = ccdb->getForTimeStamp<o2::parameters::GRPMagField>(grpmagPath, run3grp_timestamp);
+      if (!grpmag) {
+        LOG(fatal) << "Got nullptr from CCDB for path " << grpmagPath << " of object GRPMagField and " << grpPath << " of object GRPObject for timestamp " << run3grp_timestamp;
+      }
+      o2::base::Propagator::initFieldFromGRP(grpmag);
+      // Fetch magnetic field from ccdb for current collision
+      d_bz = std::lround(5.f * grpmag->getL3Current() / 30000.f);
+      LOG(info) << "Retrieved GRP for timestamp " << run3grp_timestamp << " with magnetic field of " << d_bz << " kZG";
+    }
+    mRunNumber = bc.runNumber();
+    d_bz = 0.1 * d_bz;
+  }
+
+  void process(soa::Filtered<Coll>::iterator const& collision, soa::Filtered<Trks> const& tracks, aod::BCsWithTimestamps const&)
+  {
+
+    tableRow.reserve(tracks.size());
+
+    // auto collision = collisions.begin();
+    auto bc = collision.bc_as<aod::BCsWithTimestamps>();
+    initCCDB(bc);
+
+    tableRowColl(collision.multTPC(),
+                 collision.posZ(),
+                 d_bz);
+
+    for (auto& track : tracks) {
+
+      for (auto i : particlesToReject) {
+        // if satisfied, want to continue in the upper loop (over tracks) -- skip the current track
+        // cannot use simple 'continue' since it will be applied to the current loop, so have to use 'goto'
+        if (TOFselection(track, std::make_pair(i, rejectWithinNsigmaTOF))) {
+          goto skip;
+        }
+      }
+
+      for (auto ii : particlesToKeep)
+        if (TPCselection(track, std::make_pair(ii, keepWithinNsigmaTPC))) {
+
+          tableRow(tableRowColl.lastIndex(),
+                   track.p(),
+                   track.dcaXY(),
+                   track.dcaZ(),
+                   track.tpcInnerParam(),
+                   track.tpcSignal(),
+                   track.beta(),
+                   track.tpcNClsFound(),
+                   track.tpcChi2NCl(),
+                   track.tpcCrossedRowsOverFindableCls(),
+                   track.tpcNClsShared(),
+                   track.itsNCls(),
+                   track.itsChi2NCl(),
+                   track.sign(),
+                   track.eta(),
+                   track.phi(),
+                   singletrackselector::packInTable<singletrackselector::nsigma::binning>(track.tofNSigmaPr()),
+                   singletrackselector::packInTable<singletrackselector::nsigma::binning>(track.tpcNSigmaPr()),
+                   singletrackselector::packInTable<singletrackselector::nsigma::binning>(track.tofNSigmaDe()),
+                   singletrackselector::packInTable<singletrackselector::nsigma::binning>(track.tpcNSigmaDe()));
+
+          break; // break the loop with particlesToKeep after the 'if' condition is satisfied -- don't want double entries
+        }
+
+    skip:; // here we send with 'goto'
```

Comment:
Please, no labels
Use meaningful name variables to control the execution flow

Reply:
Removed.